docs/WindowsBuild.md

```diff
@@ -0,0 +1,177 @@
+2. Open `src/win32/allinone.sln` in Visual Studio.
+3. Make sure to select the correct architecture from the drop-down in Visual
+Studio.
+4. Right click on the solution in the Solution Explorer window and select
+`Build Solution`.
+5. When this is done, add the `<icu-source>/bin` folder to your `Path`
+environment variable.
+
+### 4. Get ready
+- From within a **developer** command prompt (not PowerShell nor cmd, but [the
+  Visual Studio Developer Command
+  Prompt](https://msdn.microsoft.com/en-us/library/f35ctcxw.aspx)), execute the
+  following command if you have an x64 PC.
+```cmd
+VsDevCmd -arch=amd64
+```
+If instead, you're compiling for a 32-bit Windows target, adapt the `arch`
+argument to `x86` and run
+```cmd
+VsDevCmd -arch=x86
+```
+
+- Then adapt the following command and run it.
+```cmd
+set swift_source_dir=path-to-directory-containing-all-cloned-repositories
+```
+
+### 5. Build CMark
+- This must be done from within a developer command prompt. CMark is a fairly
+  small project and should only take a few minutes to build.
+```cmd
+mkdir "%swift_source_dir%/build/Ninja-DebugAssert/cmark-windows-amd64"
+pushd "%swift_source_dir%/build/Ninja-DebugAssert/cmark-windows-amd64"
+cmake -G "Ninja" "%swift_source_dir%/cmark"
+popd
+cmake --build "%swift_source_dir%/build/Ninja-DebugAssert/cmark-windows-amd64/"
+```
+
+### 6. Build LLVM/Clang/Compiler-RT
+- This must be done from within a developer command prompt. LLVM and Clang are
+  large projects so building might take a few hours. Make sure that the build
+  type (e.g. Debug/Release) for LLVM/Clang matches the build type for Swift.
+- Optionally, you can omit building compiler-rt by removing all lines referring
+  to `compiler-rt` below, which should give faster build times.
+```cmd
+mklink /J "%swift_source_dir%/llvm/tools/clang" "%swift_source_dir%/clang"
+mklink /J "%swift_source_dir%/llvm/tools/compiler-rt" "%swift_source_dir%/compiler-rt"
+mkdir "%swift_source_dir%/build/Ninja-RelWithDebInfoAssert/llvm-windows-amd64"
+pushd "%swift_source_dir%/build/Ninja-RelWithDebInfoAssert/llvm-windows-amd64"
+cmake -G "Ninja"^
+ -DLLVM_ENABLE_ASSERTIONS=TRUE^
+ -DCMAKE_BUILD_TYPE=RelWithDebInfo^
+ -DLLVM_TOOL_SWIFT_BUILD=NO^
+ -DLLVM_INCLUDE_DOCS=TRUE^
+ -DLLVM_TOOL_COMPILER_RT_BUILD=TRUE^
+ -DLLVM_BUILD_EXTERNAL_COMPILER_RT=TRUE^
+ -DLLVM_LIT_ARGS=-sv^
+ -DLLVM_TARGETS_TO_BUILD=X86^
+ "%swift_source_dir%/llvm"
+popd
+cmake --build "%swift_source_dir%/build/Ninja-DebugAssert/llvm-windows-amd64"
+```
+
+### 7. Build Swift
+- This must be done from within a developer command prompt and could take up to
+  two hours depending on your system.
+- You may need to adjust the `SWIFT_WINDOWS_LIB_DIRECTORY` parameter depending on
+  your target platform or Windows SDK version.
+- While the commands here use MSVC to build, using clang-cl is recommended (see
+  the **Clang-cl** section below).
+```cmd
+mkdir "%swift_source_dir%/build/Ninja-DebugAssert/swift-windows-amd64/ninja"
```

Comment:
Please remove the extra ninja directory

Reply:
doing this in a separate formatting PR, I have a list of 3-4 stylistic changes like this